java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertTextBlockToString.java

```diff
@@ -0,0 +1,106 @@
+import com.sun.source.util.TreePath;
+import org.netbeans.api.java.lexer.JavaTokenId;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.editor.hints.Fix;
+import org.netbeans.spi.java.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.java.hints.Hint;
+import org.netbeans.spi.java.hints.HintContext;
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.TriggerTreeKind;
+import org.openide.util.NbBundle.Messages;
+
+@Hint(displayName = "#DN_ConvertTextBlockToString", description = "#DESC_ConvertTextBlockToString", category = "general", minSourceVersion = "13")
+@Messages({
+    "DN_ConvertTextBlockToString=Convert Text block to String",
+    "DESC_ConvertTextBlockToString=Text Block may not be supported in older versions of java older then 13"
+})
+public class ConvertTextBlockToString {
+
+    @TriggerTreeKind(Tree.Kind.STRING_LITERAL)
+    @Messages("ERR_ConvertTextBlockToString=Text block may not be supported")//NOI18N
+    public static ErrorDescription computeWarning(HintContext ctx) {
+        TokenSequence<?> ts = ctx.getInfo().getTokenHierarchy().tokenSequence();
+        ts.move((int) ctx.getInfo().getTrees().getSourcePositions().getStartPosition(ctx.getPath().getCompilationUnit(), ctx.getPath().getLeaf()));
+        boolean moveNext = ts.moveNext();
+        if (!moveNext || ts.token().id() != JavaTokenId.MULTILINE_STRING_LITERAL) {
+            return null;
+        }
+
+        String orignalString = ctx.getPath().getLeaf().toString();
+        String orignalStringArr[] = textBlockToStringArr(orignalString.substring(1, orignalString.length() - 1));
+        Fix fix = new FixImpl(ctx.getInfo(), ctx.getPath(), orignalStringArr).toEditorFix();
+        return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_ConvertTextBlockToString(), fix);
+    }
+
+    private static String[] textBlockToStringArr(String textBlock) {
+        String oneLine[] = textBlock.split("\\\\n", -1);    // NOI18N
+        return oneLine;
+    }
+
+    private static final class FixImpl extends JavaFix {
+
+        String orignalStringArr[];
+
+        public FixImpl(CompilationInfo info, TreePath tp, String orignalStringArr[]) {
+            super(info, tp);
+            this.orignalStringArr = orignalStringArr;
+        }
+
+        @Override
+        @Messages("FIX_ConvertTextBlockToString=Convert to String")
+        protected String getText() {
+            return Bundle.FIX_ConvertTextBlockToString();
+        }
+
+        @Override
+        protected void performRewrite(TransformationContext ctx) {
+            ExpressionTree ext = buildTree(orignalStringArr, orignalStringArr.length - 1, ctx);
+            ctx.getWorkingCopy().rewrite(ctx.getPath().getLeaf(), ext);
+
+        }
+
+        static private ExpressionTree buildTree(String arr[], int li, TransformationContext ctx) {
```

Comment:
Should use proper parameter names because it is not easy to understand what `arr` and `li` mean.

Reply:
Thank you for suggestion.